### PR TITLE
feat(1131): added lint target.

### PR DESCRIPTION
**Pull request by hareshkainthmethods**
### What

Added lint target and used https://github.com/ONSdigital/dp-frontend-homepage-controller/pull/156 PR to help me to understand what I needed to modify.

### How to review

Describe the steps required to test the changes.

### Who can review

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

**Review by DavidSubiros — COMMENTED** [review]
We would also need a config file to determine what linters run, can you add a new file `.golangci.yml` to the root of the project, similar to the following link, please?

https://github.com/ONSdigital/dp-frontend-homepage-controller/pull/156/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9

Also, I think we will need to address the linter issues that happen when we run `make lint` (or you can have a look to the CI step)

File: Makefile [hunk 1]
```diff
@@ -26,13 +26,19 @@ debug:
...
+
 .PHONY: lint
-lint:
-	exit
+lint: generate-prod
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1
+	golangci-lint run ./...
 
-.PHONY: test
+generate-prod:
```
Comment (DavidSubiros) [inline Makefile:39]: Is this supposed to be a new block?

**Review by DavidSubiros — COMMENTED** [review]
LGTM, but left a question about Makefile changes.

File: Makefile (hunk 1 above)
Comment (hareshkainthmethods) [reply Makefile:39]: I think this may be a merge issue

Comment (hareshkainthmethods) [reply Makefile:40]: My mistake ive added .PHONY: test which was missing.

Comment (DavidSubiros) [reply Makefile:40]: sorry, one more very minor issue: could you also remove the 'generate-prod' line, please? (I think it's not needed, and there isn't any generate-prod step either)

**Review by DavidSubiros — APPROVED** [review]
LGTM thanks for all the changes!

**Pull request merged**